apps/openmw/mwmechanics/aisequence.cpp

```diff
@@ -96,16 +96,16 @@ std::list<AiPackage*>::const_iterator AiSequence::end() const
     return mPackages.end();
 }
 
-std::list<AiPackage*>::const_iterator AiSequence::erase(std::list<AiPackage*>::const_iterator package)
+void AiSequence::erase(std::list<AiPackage*>::const_iterator package)
 {
     // Not sure if manually terminated packages should trigger mDone, probably not?
     for(std::list<AiPackage*>::iterator it = mPackages.begin(); it != mPackages.end(); ++it)
     {
         if (package == it)
         {
-            AiPackage* packagePtr = *it;
-            delete packagePtr;
-            return mPackages.erase(it);
```

Comment:
The difference I saw between this and other similar functions in the code was that this one created a pointer (packagePtr) for the iterator value and stored it in there rather than just accessing it directly, so I think maybe cppcheck didn't like lines 106 and 107. There's not any reason to store the value in a new pointer before deleting it is there? I guess it was done for readability? Anyway I made the change for your other comment but didn't change anything here.